packages/graph-editor/src/components/panels/nodeSettings/index.tsx

```diff
@@ -81,34 +81,48 @@ const Annotations = observer(({ annotations }: Record<string, unknown>) => {
   );
 });
 
-const NodeTitle = ({ selectedNode }: { selectedNode: Node }) => {
+const NodeTitle = observer(({ selectedNode }: { selectedNode: Node }) => {
+  const [localTitle, setLocalTitle] = React.useState(
+    (selectedNode.annotations[title] as string) || '',
+  );
+
+  // Update local state when the annotation changes
+  React.useEffect(() => {
```

Comment:
The observer should trigger the rerender anyway, we shouldn't need to update the local state